velox/functions/prestosql/GeometryFunctions.h

```diff
@@ -109,6 +113,32 @@ struct StAsBinaryFunction {
   }
 };
 
+template <typename T>
+struct StPointFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE Status call(
+      out_type<Geometry>& result,
+      const arg_type<double>& x,
+      const arg_type<double>& y) {
+    if (!std::isfinite(x) || !std::isfinite(y)) {
+      return Status::UserError(fmt::format(
+          "ST_Point requires finite coordinates, got x={} y={}", x, y));
+    }
+    GEOS_TRY(
+        {
+          geos::geom::GeometryFactory::Ptr factory =
```

Comment:
I think we can reuse the factory, either making it a local variable here or have a global instance

Reply:
Put up the PR here https://github.com/facebookincubator/velox/pull/13525